daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/dfa/Rules.scala

```diff
@@ -735,3 +735,19 @@ class NLState(states: => ArrayBuffer[State], val nextState: Int, val stateNum: I
 
   def checkMatch(charIn: Char): Boolean = isNLNotCR(charIn) || isCR(charIn)
 }
+
+
+class ESState(states: => ArrayBuffer[State], val nextState: Int, val stateNum: Int)
```

Comment:
Add scaladoc making the point that this is a degenerate kind of rule that just succeeds immediately to handle the empty-string corner case.